### PR TITLE
CFE-3764 Added show/info command

### PR DESCRIPTION
**Pull request by craigcomstock**
Ticket: CFE-3764
Changelog: title

### CODE REVIEW

File: cfbs/commands.py [hunk 1]
```diff
@@ -460,5 +464,64 @@ def install_command(destination=None) -> int:
...
+
+def info_command(modules, index=None):
+    if not index:
+        index = get_index_from_config()
+    index = Index(index)
+    build = get_definition()["build"]
+    alias = None
+
+    for module in modules:
+        print()  # whitespace for readability
```
Comment (vpodzime) [inline cfbs/commands.py:480]: Should there be whitespace before the first module info?

File: cfbs/commands.py [hunk 2]
```diff
@@ -460,5 +464,64 @@ def install_command(destination=None) -> int:
...
+        if not index.exists(module):
+            print("Module '{}' does not exist".format(module))
+            continue
+        if module in index:
+            data = index[module]
+            if "alias" in data:
+                alias = module
+                module = data["alias"]
+                data = index[module]
+            found = [m for m in build if m["name"] == module]
```
Comment (vpodzime) [inline cfbs/commands.py:490]: `found = any(m for m...)`

File: cfbs/commands.py [hunk 3]
```diff
@@ -460,5 +464,64 @@ def install_command(destination=None) -> int:
...
+            data["status"] = "Added" if len(found) > 0 else "Not Added"
```
Comment (vpodzime) [inline cfbs/commands.py:491]: `if found`

File: cfbs/commands.py [hunk 4]
```diff
@@ -460,5 +464,64 @@ def install_command(destination=None) -> int:
...
+        else:
+            if not module.startswith("./"):
+                module = "./" + module
+            data = [m for m in build if m["name"] == module]
```
Comment (vpodzime) [inline cfbs/commands.py:495]: `data = next((m for m...), None)` and then `if data is None:...`

File: cfbs/commands.py [hunk 5]
```diff
@@ -460,5 +464,64 @@ def install_command(destination=None) -> int:
...
+            "status",
+            "by",
+            "tags",
+            "repo",
+            "commit",
+            "dependencies",
+            "added_by",
+            "description",
+        ]
+        data["module"] = module + "({})".format(alias) if alias else module
```
Comment (vpodzime) [inline cfbs/commands.py:513]: Make sure the `if alias else module` actually applies to the part of the expression that you want it to apply.

File: cfbs/commands.py [hunk 6]
```diff
@@ -460,5 +464,64 @@ def install_command(destination=None) -> int:
...
+            "version",
+            "status",
+            "by",
+            "tags",
+            "repo",
+            "commit",
+            "dependencies",
+            "added_by",
+            "description",
+        ]
```
Comment (vpodzime) [inline cfbs/commands.py:512]: I think this should be a tuple and it should be defined outside of this function and potentially reused in other places.

File: cfbs/commands.py [hunk 7]
```diff
@@ -460,5 +464,64 @@ def install_command(destination=None) -> int:
...
+        data["module"] = module + "({})".format(alias) if alias else module
+        for key in ordered_keys:
+            if key in data:
+                if key in ["tags", "dependencies"]:
+                    value = " ".join(data[key])
```
Comment (vpodzime) [inline cfbs/commands.py:517]: Shouldn't this be `", ".join()`?

File: cfbs/main.py [hunk 8]
```diff
@@ -85,6 +85,8 @@ def main() -> int:
         return commands.pretty_command(args.args, args.check)
     if args.command == "validate":
         return commands.validate_command(index=args.index)
+    if args.command in ["info", "show"]:
```
Comment (vpodzime) [inline cfbs/main.py:88]: tuple here, please (nitpick)

**Review by vpodzime — COMMENTED** [review]
Looks good to me otherwise. A nice addition!

File: cfbs/commands.py (hunk 7 above)
Comment (craigcomstock) [reply cfbs/commands.py:517]: Oh. I did have it that way before I refactored to operate on dependencies as well. I liked it both ways but @nickanderson preferred commas.

File: cfbs/commands.py [hunk 9]
```diff
@@ -459,3 +459,6 @@ def install_command(destination=None) -> int:
     rm(destination, missing_ok=True)
     cp("out/masterfiles", destination)
     return 0
+
+def help_command():
+    pass # no-op to allow help to be printed
```
[the remark this answers is not in the document]
Comment (craigcomstock) [reply cfbs/commands.py]: I think other parts of this mechanism could be improved but I really do like keeping it "simple" and leaving this function here. :) If you don't mind too much. I can add a better comment.

File: cfbs/commands.py [hunk 10]
```diff
@@ -459,3 +462,65 @@ def install_command(destination=None) -> int:
...
+            found = any(m for m in build if m["name"] == module)
+            data["status"] = "Added" if found else "Not Added"
+        else:
+            if not module.startswith("./"):
+                module = "./" + module
+            data = next((m for m in build if m["name"] == module), None)
+            if data is None:
+                print("Path {} exists but is not yet added as a module.".format(module))
+                continue
+            data = data[0]  # take the first (and only) item
```
Comment (vpodzime) [inline cfbs/commands.py:498]: This is wrong now, AFAICT.

**Pull request merged**